config/remote/firebase.go

```diff
@@ -63,56 +64,84 @@ func NewRConfig(updatePeriod time.Duration, rs RemoteConfigService, cm config.Ma
 func (rc *RConfig) fetchRemoteConfigIfTime() error {
 	var cfg config.Config
 	if err := rc.configManager.Load(&cfg); err != nil {
+		log.Println(internal.ErrorPrefix, "failed to load the cached config:", err)
 		return fmt.Errorf("loading config: %w", err)
 	}
 
-	if time.Now().After(cfg.RCLastUpdate.Add(rc.updatePeriod)) {
-		remoteConfigValue, err := rc.remoteService.FetchRemoteConfig()
-		if err != nil {
-			return err
-		}
-		err = json.Unmarshal(remoteConfigValue, rc.config)
-		if err != nil {
-			return err
-		}
-		if err := rc.configManager.SaveWith(func(c config.Config) config.Config {
-			s, err := json.Marshal(rc.config)
-			if err == nil {
-				c.RemoteConfig = string(s)
-				c.RCLastUpdate = time.Now()
+	if cfg.RemoteConfig == "" || time.Now().After(cfg.RCLastUpdate.Add(rc.updatePeriod)) {
```

Comment:
This is not a good change, because RC fetching will become more frequent in case of errors. Before it was a hard limit that one fetch is done per day whether it is successful or not, but now fetching will be done constantly if they are unsuccessful.